std/experimental/allocator/package.d

```diff
@@ -1099,7 +1099,11 @@ void dispose(A, T)(auto ref A alloc, T p)
 if (is(T == class) || is(T == interface))
 {
     if (!p) return;
-    auto support = (cast(void*) p)[0 .. typeid(p).initializer.length];
+    static if (is(T==interface))
+        auto ob = cast(Object) p;
```

Comment:
What if this cast returns null? I believe you can't cast C++ COM interfaces to Object (https://dlang.org/spec/interface.html#com-interfaces).